common/searchattribute/validator.go

```diff
@@ -116,7 +121,12 @@ func (v *Validator) Validate(searchAttributes *commonpb.SearchAttributes, namesp
 			)
 		}
 
-		if _, err = DecodeValue(saPayload, saType); err != nil {
+		if v.strictMode {
```

Comment:
This is the caller side I referring to in another comment. Rename `v.strictMode` to `v.allowList` and just pass it down to `DecodeValue`. I think it would be cleaner, unless I am missing something.